timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServerLauncher.java

```diff
@@ -70,4 +76,12 @@ public void run(CombinedTimeLockServerConfiguration configuration, Environment e
                 CombinedTimeLockServerConfiguration.blockingTimeoutMs(),
                 registrar);
     }
+
+    @Provider
+    private static final class EmptyOptionalTo204ExceptionMapper implements ExceptionMapper<EmptyOptionalException> {
+        @Override
+        public Response toResponse(EmptyOptionalException exception) {
+            return Response.noContent().build();
```

Comment:
seems reasonable, I guess we never had issues here because optionals were not part of the API till now